### DISCUSSION TITLE
Differences on Sentinel-2 Level2A products on PC and the ESA product. Where do they come from?

### DISCUSSION THREAD

**Discussion by RobertBehling** [microsoft/PlanetaryComputer#149] [Q&A]
Hello,

I'm happy to have the opportunity to get Sentinel-2 as level 2a data back until 2016. Therefore I compared, the level 2a data products of planetary computer (PC) with the ESA products provided at the Open Access Hub from ESA. As proposed by @TomAugspurger I will move the discussions on the data from email conversation here in the discussion part.

I have compared level 2a products before the Sentinel2 processing baseline change in Jan 2022. To conclude: the two products differ. In my example region, the PC product differs about 2% in reflectance (B08) above and below the ESA product, especially in mountainous regions. In general, the reflectance changes due to the relief are less pronounced in the ESA product. The slopes towards the sun are brighter and away the sun darker in PC compared to ESA. Please, see the attached notebook and gif. 

A part of @TomAugspurger answer from our email correspondence to the issue described above:
> Right now, my guess (just a guess) is the differences come from us using a different DEM than ESA / what's hosted on CopernicusHub. We're using[ PlanetDem](https://www.planetobserver.com/global-elevation-data), while Copernicus is using[ Copernicus DEM](https://spacedata.copernicus.eu/web/cscda/dataset-details?articleId=394198). Everything else (Sen2Cor version, auxilary datasets) appears to be the same. There are some "expert-level" parameters in Sen2Cor that can be adjusted, but we are using the defaults for all those (presumably Copernicus is as well). When I have a chance, I'll see if I can run sen2cor with Copernicus DEM to see if the results are byte-for-byte identical.

thanks @TomAugspurger for your prompt answers and I would be very interested to see how the data compare using the same DEM. 

![comparison](https://user-images.githubusercontent.com/80827055/207055916-482e7940-fecd-4b8c-bdf3-ba114704a039.gif)
[Compare_pc_esa_s2l2a_products_notebook.pdf](https://github.com/microsoft/PlanetaryComputer/files/10208356/Compare_pc_esa_s2l2a_products.pdf)

**Comment by maawoo** [comment]
It would be interesting to know why PlanetDEM was used (also for the Sentinel-1 RTC product) instead of Copernicus GLO-30 DEM. And to also see @TomAugspurger's results if he gets the chance to re-run sen2cor.

This article might be relevant: https://doi.org/10.1080/22797254.2018.1478676
> Generally, the vertical accuracy in the lowlands is much better than in the mountainous regions. In PlanetDEM, an irregular tile structure became evident, which suggests that the alpine regions are differently processed than the lowlands – probably an artefact of the reprocessing of the original SRTM and ASTER data by PlanetObserver. In some of these alpine tiles, the heights are systematically wrong by 50 m.

**Comment by mhungen** [comment]
It seems that the Planetary Computer team has [outsourced the L2A processing](https://github.com/microsoft/PlanetaryComputer/discussions/371#discussioncomment-10243439).
I think the new vendor is using the  Copernicus-DEM. 
I have compared a few new images with ESA products and the discrepancy seems to have disappeared.
